src/action.ts

```diff
@@ -259,11 +259,11 @@ class Action {
     return { baseDir: bundleDir, artifacts, executionGraph, executionGraphReport }
   }
 
-  private async extract(from: Readable, basePath: string): Promise<string[]> {
+  private async extractAdmZip(from: Readable, basePath: string): Promise<string[]> {
     const tmp = path.join(basePath, 'bundle.zip')
     const artifacts: string[] = []
     await streamPipeline(from, fs.createWriteStream(tmp))
-    await extract(tmp, { 
+    await AdmZip(tmp, { 
```

Comment:
According to [their docs](https://www.npmjs.com/package/adm-zip), it looks like the library works slightly differently to `extract-zip`. You probably want to firstly get a `zip` instance:

```js
const zip = new AdmZip(tmp);
```

And then you should be able to both extract it and get the entries (files or directories being extracted):

```js
zip.getEntries().forEach((zipEntry) =>  {
  zipEntry.entryName.endsWith(path.sep) ? null : artifacts.push(path.join(basePath, zipEntry.entryName))
});
zip.extractAllTo(basePath, true);
```